gcf_data_mapper/parsers/family.py

```diff
@@ -3,6 +3,132 @@
 import click
 import pandas as pd
 
+from gcf_data_mapper.enums.family import (
+    FamilyColumnsNames,
+    FamilyNestedColumnNames,
+    GCFProjectBudgetSource,
+)
+from gcf_data_mapper.parsers.helpers import (
+    arrays_contain_empty_values,
+    check_row_for_missing_columns,
+    row_contains_columns_with_empty_values,
+)
+
+
+def get_budgets(funding_list: list[dict], source: str) -> list[int]:
+    """Get the budget amount from the row based on the funding source.
+
+    :param list[dict] row: A list of all the funding information, represented in dictionaries
+    :param str source: The funding source to retrieve the budget from.
+
+    :return list[int]: A list of budget amounts corresponding to the source,
+        or [0] if the source is not found.
+    """
+
+    budget_key = FamilyNestedColumnNames.BUDGET.value
+    source_key = FamilyNestedColumnNames.SOURCE.value
+
+    budgets = [
+        funding[budget_key] for funding in funding_list if funding[source_key] == source
+    ]
+
+    return budgets if budgets else [0]
```

Comment:
Magic number - why 0? What is 0 in this context?

Reply:
Peter and I had a discussion about how to return funding info when it was only financed by GCF or alternatively only cofinanced, so where there would be no `project_value_co_financing` values for example, he said in these instances return [0].

Reply:
Can you put this as a comment inside the code then please, as that's key context :)